### DISCUSSION TITLE
Follow NEP29 drop schedule or not?

### DISCUSSION THREAD

**Discussion by tkoyama010** [pyvista/pyvista#4960] [Polls]
We were talking at https://github.com/pyvista/pytest-pyvista/pull/85 about whether the PyVista project should follow [NEP29 drop schedule](https://numpy.org/neps/nep-0029-deprecation_policy.html#drop-schedule). Right now we continue to support each version of Python until it is EOL, but if we follow NEP29, the end of support may be sooner than that. Please let me know what you all think.

**Comment by MatthewFlamm** [comment]
> This NEP is superseded by the scientific python ecosystem coordination 
guideline SPEC 0 — Minimum Supported Versions.

If the goal of this suggestion is to align with other scientific python packages, we should be following SPEC 0. https://scientific-python.org/specs/spec-0000/ According to SPEC 0, python 3.9 would be dropped in just over a week.

My main concern is that vtk does not adhere to this support window as far as I know.  vtk's most recent 9.2.6 still supports Python 3.6 as does the rc for 9.3.0(!).  What numpy versions does vtk intend to support?  I think this sort of information should guide the discussion here.
